portfolio/src/main/java/com/google/sps/servlets/HandleCommentsServlet.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/*
+ *  Servlet that adds comments
+ */
+@WebServlet("/handle-comments")
+public class HandleCommentsServlet extends HttpServlet {
+
+  @Override
+  public void init() {}
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = request.getParameter("add-content");
+    String color = request.getParameter("color");
+    String font_size = request.getParameter("font_size");
+    long time = System.currentTimeMillis();
+
+    UserService userService = UserServiceFactory.getUserService();
+    String username = userService.getCurrentUser().getNickname();
```

Comment:
It would be best to check `isUserLoggedIn()` here before calling `getCurrentUser()`. This code throws a NullPointerException if no user is logged in. If the user is indeed not logged in, then you could return a not-ok status with `response.setStatus(HttpServletResponse.SC_FORBIDDEN);`. SC_FORBIDDEN is "403 - permission denied" I'm not sure that's the most appropriate one here, there might be a better option.

I know that your frontend code is structured so that the user can't click on the `add content` button unless they are logged in, but this is a pattern that should be avoided. The general rule is: **you can't trust the frontend**, since the frontend code runs on the user side, and they can mess with it :)
In this case, issuing a `handle-comments` post request without being logged in would crash this servlet.